nsv13/code/modules/overmap/fighters/fighters_components.dm

```diff
@@ -265,24 +265,274 @@
+		fuel_target = null
+		STOP_PROCESSING(SSobj, src)
+	if(action == "transfer_mode")
+		if(!allow_refuel)
+			to_chat(usr, "<span class='notice'>You open [src]'s fuel inlet valve, it will now intake reagents from containers that it's hit with.</span>")
+			allow_refuel = TRUE
+		else
+			to_chat(usr, "<span class='notice'>You close [src]'s fuel inlet valve, it will now transfer its reagents to containers that it's hit with.</span>")
+			allow_refuel = FALSE
+
+/obj/structure/reagent_dispensers/fueltank/aviation_fuel/can_interact(mob/user)
+	if(last_user == user) //If theyre holding the hose, bypass range checks so that they can see what they're currently fuelling.
+		return TRUE
+	if(!user.can_interact_with(src)) //Theyre too far away and not flying the ship
+		return FALSE
+	if((interaction_flags_atom & INTERACT_ATOM_REQUIRES_DEXTERITY) && !user.IsAdvancedToolUser())
+		to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
+		return FALSE
+	if(!(interaction_flags_atom & INTERACT_ATOM_IGNORE_INCAPACITATED) && user.incapacitated((interaction_flags_atom & INTERACT_ATOM_IGNORE_RESTRAINED), !(interaction_flags_atom & INTERACT_ATOM_CHECK_GRAB)))
+		return FALSE
+	return TRUE
+
+/obj/structure/reagent_dispensers/fueltank/aviation_fuel/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state) // Remember to use the appropriate state.
+	if(!can_interact(user))
+		return
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "aviation_fuel", name, 400, 400, master_ui, state)
+		ui.open()
+
+/obj/structure/reagent_dispensers/fueltank/aviation_fuel/ui_data(mob/user)
+	var/list/data = list()
+	if(!fuel_target)
+		data["targetfuel"] = 0
+		data["targetmaxfuel"] = 1000 //Default value to black out the fuel gauge.
+	else
+		data["targetfuel"] = fuel_target.get_fuel()
+		data["targetmaxfuel"] = fuel_target.get_max_fuel()
+	data["transfer_mode"] = allow_refuel
+	data["fuel"] = reagents.total_volume
+	data["maxfuel"] = reagents.maximum_volume
+	return data
+
+/obj/structure/reagent_dispensers/fueltank/aviation_fuel/Initialize()
+	. = ..()
+	add_overlay("jetfuel_nozzle")
+	nozzle = new(src)
+	nozzle.parent = src
+	soundloop = new(list(src), FALSE)
+	reagents.flags |= REFILLABLE
+
+/obj/structure/reagent_dispensers/fueltank/aviation_fuel/attack_hand(mob/user)
+	if(nozzle.loc == src)
+		if(!user.put_in_hands(nozzle))
+			to_chat(user, "<span class='warning'>You need a free hand to hold the fuel hose!</span>")
+			return
+		to_chat(user, "<span class='warning'>You grab [src]'s refuelling hose.</span>")
+		last_user = user
+		RegisterSignal(last_user, COMSIG_MOVABLE_MOVED, .proc/check_distance)
+		toggle_nozzle(FALSE)
+		ui_interact(user)
+	else
+		ui_interact(user)
+		. = ..()
+
+/obj/effect/ebeam/fuel_hose
+	name = "fuel hose" //Nsv13 - Modified beam.dmi to include a fuel hose.

```

Comment:
Ideally it should be in a NSV beam file and redirected at this location to there for merging newer stuff from elsewhere. I said that helps with merging from upstream before remember.

Reply:
Aka I dont know how to fix dmi conflicts.